### PR TITLE
solution added

### PR DESCRIPTION
**Pull request by Vosker**

### CODE REVIEW

**Review by voloshynamaryna11 — COMMENTED** [review]
nice

File: src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(cinemaHall);
+            transaction.commit();
+            return cinemaHall;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Failed to add the cinema hall", e);
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java]: let's make message more informative and add some info about entity we were trying to save same with other methods, where applicable

File: src/main/java/com/dev/cinema/model/MovieSession.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+@Entity
+@Table(name = "movie_sessions")
+public class MovieSession {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
+    private Movie movie;
+    @ManyToOne
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/model/MovieSession.java:19]: this is for your info, but keep in mind that toOne relations have fetchType eager by default, which is not always quite good

**Review by RomanDubovskyi — COMMENTED** [review]
good overall, just make exception messages more informative

**Comment by RomanDubovskyi** [comment]
travis failed

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**